web/react/components/filtered_user_list.jsx

```diff
@@ -48,6 +92,7 @@ class FilteredUserList extends React.Component {
 
         let users = this.props.users;
 
+        // this logic might be better placed in UserStore
```

Comment:
This is actually the right place for this logic. Derived state like this is supposed to be handled in the render function since it solely depends on the filter state and user props. There's no reason to store the filtered users since that will only change when either the filter or users change